src/client/src/database.rs

```diff
@@ -114,22 +122,33 @@ impl Database {
         self.handle(Request::Inserts(requests)).await
     }
 
+    pub async fn insert_to_stream(&self, requests: InsertRequests) -> Result<()> {
+        let streaming_client = {
+            let mut streaming_client = self.streaming_client.lock().await;
+            if streaming_client.is_none() {
+                *streaming_client = Some(self.client_stream().await?);
+            }
+            streaming_client.as_ref().unwrap().clone()
+        };
```

Comment:
I suggest to use `tokio::sync::OnceCell`:

```rust
streaming_client: OnceCell<Sender<GreptimeRequest>>,
```

Then you can simply:

```suggestion
        let streaming_client = streaming_client.get_or_try_init(self.client_stream())?.clone();
```